bundles/mapping/mapmodule/oskariStyle/generator.ol.js

```diff
@@ -1,15 +1,223 @@
+    defaultStyles[layerType] = styleDef;
+};
+const getDefaultStyle = layerType => {
+    return defaultStyles[layerType] || {};
+};
+
+const getFeatureStyle = (layer, extendedDef = {}) => {
+    const style = layer.getCurrentStyle();
+    const defaultDef = getDefaultStyle(layer.getLayerType());
+    return merge(defaultDef, style.getFeatureStyle(), extendedDef);
+};
+
+export const useStyleFunction = layer => {
+    const styleType = geometryTypeToStyleType(layer.getGeometryType());
+    const hasOptionalStyles = layer.getCurrentStyle().getOptionalStyles().length > 0;
+    const hasCluster = typeof layer.getClusteringDistance() !== 'undefined';
+    return hasOptionalStyles || hasCluster ||
+        !styleType || styleType === STYLE_TYPE.COLLECTION;
+};
+
+export const geometryTypeToStyleType = type => {
+    let styleType;
+    switch (type) {
+    case 'LineString':
+    case 'MultiLineString':
+        styleType = STYLE_TYPE.LINE; break;
+    case 'Polygon':
+    case 'MultiPolygon':
+        styleType = STYLE_TYPE.AREA; break;
+    case 'Point':
+    case 'MultiPoint':
+        styleType = STYLE_TYPE.POINT; break;
+    case 'GeometryCollection':
+        styleType = STYLE_TYPE.COLLECTION; break;
+    }
+    return styleType;
+};
+
+export const getOlStyleForLayer = (mapmodule, layer, extendedDef) => {
+    const featureStyle = getFeatureStyle(layer, extendedDef);
+    if (!useStyleFunction(layer)) {
+        const styleType = geometryTypeToStyleType(layer.getGeometryType());
+        return mapmodule.getStyle(featureStyle, styleType);
+    }
+    const typed = mapmodule.getGeomTypedStyles(featureStyle);
+    const optional = layer.getCurrentStyle().getOptionalStyles().map(optionalDef => {
+        return {
+            filter: getOptionalStyleFilter(optionalDef),
+            typed: mapmodule.getGeomTypedStyles(merge(featureStyle, optionalDef))
+        };
+    });
+    return getStyleFunction({ typed, optional });
+};
+
+export const getStyleForGeometry = (geometry, styleTypes) => {
+    if (!geometry || !styleTypes) {
+        return;
+    }
+    let geometries = [];
+    const geomType = geometry.getType ? geometry.getType() : geometry;
+    const type = geometryTypeToStyleType(geomType);
+    if (type === STYLE_TYPE.COLLECTION) {
+        if (typeof geometry.getGeometries === 'function') {
+            geometries = geometry.getGeometries() || [];
+        }
+        if (geometries.length > 0) {
+            log.debug('Received GeometryCollection. Using first feature to determine feature style.');
+            return getStyleForGeometry(geometries[0], styleTypes);
+        } else {
+            log.info('Received GeometryCollection without geometries. Feature style cannot be determined.');
+        }
+    };
+    return styleTypes[type];
+};
+
+// Style for cluster circles
+const clusterStyleCache = {};
+const clusterStyleFunc = feature => {
+    const size = feature.get('features').length;
+    const cacheKey = [`${size}`];
```

Comment:
Cache key is an array? Should be able to use just as string. And size is really only 14 or 12 for radius so we could just do:
```suggestion
    const radius = size > 9 ? 14 : 12;
    const cacheKey = `${radius }`;
```